### PR TITLE
Fix building pip packages.

### PR DESCRIPTION
**Pull request by csukuangfj**
Now k2 can be installed by

```bash
  python3 setup.py install
```

---

I have removed `scripts/build_pip.sh`.

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -17,22 +50,72 @@
...
+        # build/lib.linux-x86_64-3.8
+        os.makedirs(self.build_lib, exist_ok=True)
+
+        k2_dir = os.path.dirname(os.path.abspath(__file__))
+
+        cmake_args = os.environ.get('K2_CMAKE_ARGS', '')
+        make_args = os.environ.get('K2_MAKE_ARGS', '')
+
+        if cmake_args == '':
+            cmake_args = '-DCMAKE_BUILD_TYPE=Release'
```
Comment (pzelasko) [inline setup.py:93]: I think it was already suggested somewhere, but wouldn't `RelWithDebInfo` help solving issues reported by the users by giving more informative stack traces / gdb / cuda-gdb information?

Comment (csukuangfj) [reply setup.py:93]: It was mentioned in https://github.com/k2-fsa/k2/pull/734#discussion_r629095496

> I am not sure whether it's worthwhile to change -O3 to -O2. If users want to report issues, I would
recommend them to use Debug.

I often see projects are released with `Release` build, not `RelWithDebInfo` build.

Comment (pzelasko) [reply setup.py:93]: Alright, missed that one. Cool, makes sense.

**Comment by danpovey** [comment]
LGTM!  (Not thorough check...)

**Comment by csukuangfj** [comment]
Will merge in the morning.

**Pull request merged**